### PR TITLE
[FLINK-15446][table][docs] Improve "Connect to External Systems" documentation page

### PR DESCRIPTION
**Pull request by wuchong**
## What is the purpose of the change

Further improve "Connect to External Systems" page under Table API & SQL. 

## Brief change log

1. Deprecate schema methods for CSV and JSON formats in API
2. Remove documentation for format schema, which is not necessary any more and is deprecated.
3. Add DDL documentation for "Table Schema" and "Rowtime Attribute" sections.
4. Update the comments in DDL for better rendering (do not wrap line).

## Verifying this change

No tests. 

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by wuchong** [comment]
cc @JingsongLi

File: docs/dev/table/connect.md [hunk 1]
```diff
@@ -164,21 +164,8 @@ CREATE TABLE MyUserTable (
   'connector.properties.zookeeper.connect' = 'localhost:2181',
   'connector.properties.bootstrap.servers' = 'localhost:9092',
 
-  -- specify the update-mode for streaming tables
-  'update-mode' = 'append',
-
   -- declare a format for this system
-  'format.type' = 'avro',
```
Comment (JingsongLi) [inline docs/dev/table/connect.md]: Does Avro support derive schema from connector?

File: docs/dev/table/connect.md [hunk 2]
```diff
@@ -371,9 +335,23 @@ schema:
...
-For *each field*, the following properties can be declared in addition to the column's name and type:
+In order to declare time attributes in the schema, the following ways are supported:
 
 <div class="codetabs" markdown="1">
+<div data-lang="DDL" markdown="1">
+{% highlight sql %}
+CREATE TABLE MyTable (
+  MyField1 AS PROCTIME(), -- declares this field as a processing-time attribute
+  MyField2 TIMESTAMP(3),
+  MyField3 BOOLEAN,
```
Comment (JingsongLi) [inline docs/dev/table/connect.md]: Looks like `MyField3` not same to python and Java/Scala.
Can we do this rename by compute column?

File: docs/dev/table/connect.md [hunk 3]
```diff
@@ -426,9 +404,32 @@ For more information about time handling in Flink and especially event-time, we
...
+<div data-lang="DDL" markdown="1">
+{% highlight sql %}
+-- use the existing TIMESTAMP(3) field in schema as the rowtime attribute
+CREATE TABLE MyTable (
+  ts_field TIMESTAMP(3),
+  WATERMARK FOR ts_field AS ...
+) WITH (
+  ...
+)
+
```
Comment (JingsongLi) [inline docs/dev/table/connect.md]: Do we support `timestampsFromSource`?

File: docs/dev/table/connect.md [hunk 4]
```diff
@@ -500,6 +501,39 @@ rowtime:
...
+-- Sets a watermark strategy for ascending rowtime attributes. Emits a watermark of the maximum
+-- observed timestamp so far minus 1. Rows that have a timestamp equal to the max timestamp
+-- are not late.
+CREATE TABLE MyTable (
+  ts_field TIMESTAMP(3),
+  WATERMARK FOR ts_field AS ts_field - INTERVAL '0.001' SECOND
```
Comment (JingsongLi) [inline docs/dev/table/connect.md]: Why not same to Java/Scala?

File: docs/dev/table/connect.md [hunk 5]
```diff
@@ -500,6 +501,39 @@ rowtime:
...
+) WITH (
+  ...
+)
+
+-- Sets a watermark strategy for rowtime attributes which are out-of-order by a bounded time interval.
```
Comment (JingsongLi) [inline docs/dev/table/connect.md]: It is same to above? Why are out-of-order?

File: docs/dev/table/connect.md (hunk 1 above)
Comment (wuchong) [reply docs/dev/table/connect.md]: No, we don't support yet. That's why I change the example to json which is simpler.

File: docs/dev/table/connect.md (hunk 2 above)
Comment (wuchong) [reply docs/dev/table/connect.md]: There is a little differentce between using computed column and `Schema#from()`. Using `Schema#from()`, users can't  reference the original field name.

File: docs/dev/table/connect.md (hunk 3 above)
Comment (wuchong) [reply docs/dev/table/connect.md]: We don't support yet.

File: docs/dev/table/connect.md (hunk 4 above)
Comment (wuchong) [reply docs/dev/table/connect.md]: We don't have a shortcut API for `watermarksPeriodicAscending` in SQL DDL.

File: docs/dev/table/connect.md (hunk 5 above)
Comment (wuchong) [reply docs/dev/table/connect.md]: We don't have a shortcut API for this. 
All the watermark strategies are expressed by expressions, we didn't provide built-in strategy functions. So the `asceding timestamp` is a special case of `out-of-order timestamp` which has a constant 1 ms delay.

File: docs/dev/table/connect.md (hunk 2 above)
Comment (JingsongLi) [reply docs/dev/table/connect.md]: Can we do this rename by compute column?

File: docs/dev/table/connect.md [hunk 6]
```diff
@@ -164,21 +164,8 @@ CREATE TABLE MyUserTable (
   'connector.properties.zookeeper.connect' = 'localhost:2181',
   'connector.properties.bootstrap.servers' = 'localhost:9092',
```
Comment (JingsongLi) [inline docs/dev/table/connect.md]: You need modify above "avro".

File: docs/dev/table/connect.md [hunk 7]
```diff
@@ -164,21 +164,8 @@ CREATE TABLE MyUserTable (
...
 
-  -- specify the update-mode for streaming tables
```
Comment (JingsongLi) [inline docs/dev/table/connect.md]: Change above "The connector might already provide a fixed format with fields and schema." to "The connector might already provide a fixed format."?

Comment (JingsongLi) [inline docs/dev/table/connect.md]: And above "For streaming queries, an update mode declares how to communicate between a dynamic table and the storage system for continuous queries."
need remove or not?

File: docs/dev/table/connect.md (hunk 3 above)
Comment (JingsongLi) [reply docs/dev/table/connect.md]: Can we add a comment to tell user? Because the DDL is the first tab now.

File: docs/dev/table/connect.md (hunk 5 above)
Comment (JingsongLi) [reply docs/dev/table/connect.md]: It is same to above example... Why different description?

File: docs/dev/table/connect.md (hunk 4 above)
Comment (JingsongLi) [reply docs/dev/table/connect.md]: Can we add a example for how to write `watermarksPeriodicAscending`?

File: docs/dev/table/connect.md (hunk 2 above)
Comment (wuchong) [reply docs/dev/table/connect.md]: Yes, we can.

File: docs/dev/table/connect.md (hunk 4 above)
Comment (wuchong) [reply docs/dev/table/connect.md]: This is the example.

File: docs/dev/table/connect.md (hunk 5 above)
Comment (wuchong) [reply docs/dev/table/connect.md]: The difference is that `ascending timestamp` must use `INTERVAL '0.001' SECOND`.

Comment (JingsongLi) [reply docs/dev/table/connect.md]: Can we change to 2s? We can keep same to Java/Scala/Python.

File: docs/dev/table/connect.md (hunk 4 above)
Comment (JingsongLi) [reply docs/dev/table/connect.md]: I see

File: docs/dev/table/connect.md (hunk 7 above)
Comment (wuchong) [reply docs/dev/table/connect.md]: I would like to add a following sentence, what do you think?

The connector might already provide a default update mode (e.g. Kafka connector works in append mode by default).

**Comment by wuchong** [comment]
@JingsongLi  PR is update.

**Comment by wuchong** [comment]
Thanks @JingsongLi  for the reviewing. 
Merging...

**Pull request closed without merging**